### ISSUE TITLE
Different UI interface from Windows Setup when using Rufus 3.19

### ISSUE THREAD

**Issue by Wozmar** [pbatard/rufus#1971]
Why Rufus 3.19 changes the windows 11 boot window after applying the changes.
The original boot image is pictures 1 and 2
After applying the changes, there is no option from photo 2 "Repair your computer"
There are photo windows 4, 5 and 6.
How to restore windows from photos 1 and 2 after applying changes by Rufus 3.19

![1](https://user-images.githubusercontent.com/108876976/177773767-f361093e-58a8-4c8b-9dfa-b1e639d8d58c.png)
![2](https://user-images.githubusercontent.com/108876976/177773778-50252e22-f8b0-4619-a893-82d9939a4e29.png)
![3](https://user-images.githubusercontent.com/108876976/177773781-a33d72ab-6ba0-49d4-ab4f-c85f94b89225.PNG)
![4](https://user-images.githubusercontent.com/108876976/177773784-fca985e4-15d8-47de-b312-5219c596a2fd.PNG)
![5](https://user-images.githubusercontent.com/108876976/177773787-097c1cbf-6f9c-4d6f-ba8b-d73980984206.PNG)

**Comment by pbatard** [comment]
Because this is something I was pondering about doing, mostly because you also get a command prompt briefly appearing while the bypass for Secure Boot/TMP/Disk/RAM are applied to the registry, I have now restored the ability for Rufus to patch the registry keys directly and disable the `windowsPE` phase from `unattend.xml`. This should restore the setup UI to the default one.

Note however that this will only work for the regular version of Rufus and not the Windows Store version (which is the reason we change the way we inserted the registry keys in 3.19 in the first place, because Microsoft restricts the ability of Store apps from editing an offline registry).

**Comment by Wozmar** [comment]
Rufus 3.18 does not change the original boot image when selecting Extended Windows 11 Installation (no TPM / no Secure Boot).

**Comment by pbatard** [comment]
That's what I'm saying. Again, _"the reason (why) we changed the way we inserted the registry keys in 3.19 in the first place (is) because Microsoft restricts the ability of Store apps from editing an offline registry"_ which implies that Rufus 3.18 did not change it, since, as I also explained _"Microsoft (decides if it) shows one type of screen or the other according to whether it detects a windowsPE pass in the unattend.xml"_ which is the part we added to 3.19.

**Comment by Wozmar** [comment]
Thank you for your answer for me the original UI from Windows Installer is very important in VMware I am doing update integrations in SysPrep state and I need to have Repair your computer option.
Sorry for my english but I use an interpreter.
I think Rufus 3.20 will be better.
W dniu 2022-07-09 05:09:05 użytkownik Pete Batard ***@***.***> napisał:
 
That's what I'm saying. Again, "the reason (why) we changed the way we inserted the registry keys in 3.19 in the first place (is) because Microsoft restricts the ability of Store apps from editing an offline registry" which implies that Rufus 3.18 did not change it, since, as I also explained "Microsoft (decides if it) shows one type of screen or the other according to whether it detects a windowsPE pass in the unattend.xml" which is the part we added to 3.19.

**Comment by pbatard** [comment]
Well, because of new issue #1981, it looks like I'm going to be forced to revert the workaround I was planning to apply in Rufus 3.20 in order to restore the default initial Windows Setup screens, because it turns out that it is not possible to bypass the MSA requirement of Windows 11 without also having a `windowsPE` pass in `unattend.xml`, and as soon as you have a `windowsPE` pass in `unattend.xml`, then the Windows Setup screen are changed...

In short, the only way you'll be able to get the default UI screen is you make unselect ALL the bypass options when creating the media, because I currently see no cost/effective solution where we can avoid using a `windowsPE` pass altogether (especially if I also want to propose the option to automatically select locale/country to the same as the PC on which the media was created).

I'm still looking, but I'm really not optimistic at this stage...
